src/chain.h

```diff
@@ -380,7 +380,7 @@ class CDiskBlockIndex : public CBlockIndex
 
         // Only read/write nSproutValue if the client version used to create
         // this index was storing them.
-        if ((nType & SER_DISK) && (nVersion >= SPROUT_VALUE_VERSION)) {
+        if ((s.GetType() & SER_DISK) && (nVersion >= SPROUT_VALUE_VERSION)) {
```

Comment:
``nVersion`` here refers to ``s.GetVersion()``, which may be different from the block header version ``this->nVersion``. I suggest replacing ``nVersion`` with ``s.GetVersion()`` here and on line 346 to make it more explicit.

Reply:
That is, although the local doesn't *technically* shadow the member variable, there is a way to refer to the member variable (as just ``nVersion``) that would usually be correct and here is not. That's confusing.